cmd/distributor/main.go

```diff
@@ -48,6 +48,8 @@ func main() {
 	if err != nil {
 		log.Fatalf("Error initializing distributor: %v", err)
 	}
+	go dist.Run()
```

Comment:
Ah right. Totally missed that these new jobs now also use the distributor internally. Of course. Will fix.

In the past (in Prometheus) we've usually settled on leaving concurrency up to the caller, also in case of these kinds of constructors. The ruler does it like that too, but e.g. the ingester doesn't. I'm cool either way. I'll change the `go` to be in the constructor for now?